consensus/src/dag/dag_handler.rs

```diff
@@ -150,11 +150,11 @@ impl NetworkHandler {
                     .to_bytes(&response_msg.into_network_message())
                     .map(Bytes::from)
             })
-            .map_err(RpcError::ApplicationError);
+            .map_err(RpcError::ApplicationError)?;
```

Comment:
Hmm, I actually wanted to send the response back to the sender as well. Otherwise, the remote rpc might keep retrying after timing out. 

I wanted to distinguish between retriable (parent missing) vs non-retriable errors so sender can make a retry decision accordingly.

Reply:
our rpc doesn't work like this, it just sends back to network layer and logs there. if we want to send back to sender, we need to add an enum for error message

Reply:
where does it log? receiver side or sender side?

Reply:
locally

Reply:
It seems like the rpc receiver side is sending back the whole result? https://github.com/aptos-labs/aptos-core/blob/e7f32fbd72bae6d1a9b334e0b27e4aff68d9ec04/network/framework/src/peer/mod.rs#L272

Reply:
nvm, I see the error result getting dropped later.